File: examples/java/pact_validation_with_oas/one_consumer_one_provider/consumer/src/main/java/br/com/zup/pact/consumer/repository/ClientRepository.java
package br.com.zup.pact.consumer.repository;

import br.com.zup.pact.consumer.dto.ClientDetailsDTO;

import java.util.List;
import java.util.Optional;

public interface ClientRepository {
    Optional<List<ClientDetailsDTO>> getAll();
    Optional<ClientDetailsDTO> findByClientId(Integer clientId);
}

File: example/java/kotlin/consumer/src/main/kotlin/br/com/zup/pact/consumer/repository/ClientRepository.kt
package br.com.zup.pact.consumer.repository

import br.com.zup.pact.consumer.dto.ClientDetailsDTO
import br.com.zup.pact.consumer.stub.ClientStub
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import java.util.*

@Component
class ClientRepository(@Autowired val clientStub: ClientStub) {

    fun findByClientId(clientId: Int): Optional<ClientDetailsDTO> {
        return Optional.ofNullable(
                clientStub.clients[clientId]?.toAccountDetailsDTO()
        )
    }

    fun getAll(): List<ClientDetailsDTO> {
        return clientStub.getAllStubsDTOFormat()
    }

}